### PR TITLE
Added more Python resources, updated links/info

### PR DESCRIPTION
**Pull request by learnbyexample**
## What does this PR do?
Add resource(s) | Improve repo

### Description

* Added more Python resources.
* Changed http to https, added/updated author/book info for some of the resources that I'm familiar with.

### Why is this valuable (or not)?

I've glanced through these resources and found them to be useful.

### How do we know it's really free?

I've checked the links before adding them.

### For book lists, is it a book? For course lists, is it a course? etc.

Newly added links are all books.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists  in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Followup

- Check the output of Travis-CI for linter errors!

### CODE REVIEW

File: books/free-programming-books.md [hunk 1]
```diff
@@ -2491,10 +2494,12 @@ Kerridge (PDF) (email address *requested*, not required)
 * [Picking a Python Version: A Manifesto](https://www.oreilly.com/ideas/picking-a-python-version) - David Mertz
 * [Porting to Python 3: An In-Depth Guide](http://python3porting.com) (2.6 - 2.x & 3.1 - 3.x)
 * [Practical Programming in Python](https://launchpadlibrarian.net/165489933/PracticalProgrammingPython2014.pdf) - Jeffrey Elkner (PDF)
+* [Practical Python Programming](https://dabeaz-course.github.io/practical-python/) - David Beazley (3.6)
```
Comment (eshellman) [inline books/free-programming-books.md:2497]: The author describes this as a course, so we should list it a a course.

File: books/free-programming-books.md [hunk 2]
```diff
@@ -2570,6 +2576,7 @@ Kerridge (PDF) (email address *requested*, not required)
 
 * [Learn Pandas](https://bitbucket.org/hrojas/learn-pandas) - Hernan Rojas (0.18.1)
 * [pandas: powerful Python data analysis toolkit](https://pandas.pydata.org/docs/pandas.pdf) - Wes McKinney and the Pandas Development Team (:construction: *in process*) (PDF)
+* [Python Pandas For Your Grandpa](https://www.gormanalysis.com/blog/python-pandas-for-your-grandpa/) - Ben Gorman
```
Comment (eshellman) [inline books/free-programming-books.md:2579]: Also a course.

**Review by eshellman — CHANGES_REQUESTED** [review]
These are outstanding additions! The two courses just need to be moved to the courses list.

**Comment by learnbyexample** [comment]
Sorry about that! I've removed those two resources, will make a separate courses PR later.

**Pull request merged**